File: 프로그래머스/0/181836. 그림 확대/그림 확대.java
import java.util.*;

class Solution {
    public String[] solution(String[] picture, int k) {
        List<String> result = new ArrayList<>();
        for (String row : picture) {
            StringBuilder newRow = new StringBuilder();
            for (char c : row.toCharArray()) {
                for (int i = 0; i < k; i++) {
                    newRow.append(c);
                }
            }
            String repeatedRow = newRow.toString();
            for (int i = 0; i < k; i++) {
                result.add(repeatedRow);
            }
        }
        return result.toArray(new String[0]);
    }
}

File: 프로그래머스/0/181836. 그림 확대/그림 확대.js
function solution(picture, k) {
    const result = [];
    for (let row of picture) {
        const newRow = row.split('').map(c => c.repeat(k)).join('');
        for (let i = 0; i < k; i++) {
            result.push(newRow);
        }
    }
    return result;
}
